### ISSUE TITLE
设置 ProviderConfig的export 属性无效

### ISSUE THREAD

**Issue by gentryhuang** [apache/dubbo#7012]
### Environment

* Dubbo version: 2.7.7
* Java version: 1.8

### Steps to reproduce this issue

1. config provider
```xml
  <dubbo:provider  export="false"/>
```

2. ServiceConfigBase

```java 
--- ServiceConfigBase

  protected ProviderConfig provider;  // null

  @Override
    public Boolean getExport() {
       // provider == null
        return (export == null && provider != null) ? provider.getExport() : export;
    }
```

### Expected Result

```java 
    public synchronized void export() {
        if (!shouldExport()) {  // expected true
            return;
        }
}
```

### Actual Result
```java 
    public synchronized void export() {
        if (!shouldExport()) {  //  return false
            return;
        }
```

**Comment by CarefulHuo** [comment]
在这里蹲一个大佬的回答

**Comment by CarefulHuo** [comment]
言语就在嘴边，奈何就是说不出来

**Comment by xiaoheng1** [comment]
I don't quite understand this, can you provide a complete example? When I tested here, I found export = false

**Comment by gentryhuang** [comment]
> I don't quite understand this, can you provide a complete example? When I tested here, I found export = false

 I set up `<dubbo:provider  export="false"/>`, the goal is not to expose the service，but the service is still exposed。If i set up  `<dubbo:service export="false"/>`,this is ok。

**Comment by xiaoheng1** [comment]
config:

```
<dubbo:protocol name="dubbo" port="-1"/>

    <dubbo:provider export="false" />

    <bean id="demoService" class="org.apache.dubbo.demo.provider.DemoServiceImpl"/>
    <bean id="greetingService" class="org.apache.dubbo.demo.provider.GreetingServiceImpl"/>

    <dubbo:service interface="org.apache.dubbo.demo.DemoService" timeout="3000" ref="demoService" registry="registry1"/>
    <dubbo:service version="1.0.0" group="greeting" timeout="5000" interface="org.apache.dubbo.demo.GreetingService"
                   ref="greetingService"/>
```

I found that there is no exposed service, but the container is started

**Comment by gentryhuang** [comment]
> config:
> 
> ```
> <dubbo:protocol name="dubbo" port="-1"/>
> 
>     <dubbo:provider export="false" />
> 
>     <bean id="demoService" class="org.apache.dubbo.demo.provider.DemoServiceImpl"/>
>     <bean id="greetingService" class="org.apache.dubbo.demo.provider.GreetingServiceImpl"/>
> 
>     <dubbo:service interface="org.apache.dubbo.demo.DemoService" timeout="3000" ref="demoService" registry="registry1"/>
>     <dubbo:service version="1.0.0" group="greeting" timeout="5000" interface="org.apache.dubbo.demo.GreetingService"
>                    ref="greetingService"/>
> ```
> 
> I found that there is no exposed service, but the container is started

config-1:
```xml
    <dubbo:provider export="false"/>
    <dubbo:service interface="com.code.dubbo.demo.DemoService" ref="demoServiceImpl"/>
```
result-1:
```text
[DUBBO] Export dubbo service com.code.dubbo.demo.DemoService to local registry url : injvm://127.0.0.1/com.code.dubbo.demo.DemoService?accesslog=true&....
```

config-2:
```xml
     <dubbo:provider export="false"/>
    <dubbo:service interface="com.code.dubbo.demo.DemoService" ref="demoServiceImpl" export="false"/>
```
result-2:
```text
not export
```

**Comment by gentryhuang** [comment]
> config:
> 
> ```
> <dubbo:protocol name="dubbo" port="-1"/>
> 
>     <dubbo:provider export="false" />
> 
>     <bean id="demoService" class="org.apache.dubbo.demo.provider.DemoServiceImpl"/>
>     <bean id="greetingService" class="org.apache.dubbo.demo.provider.GreetingServiceImpl"/>
> 
>     <dubbo:service interface="org.apache.dubbo.demo.DemoService" timeout="3000" ref="demoService" registry="registry1"/>
>     <dubbo:service version="1.0.0" group="greeting" timeout="5000" interface="org.apache.dubbo.demo.GreetingService"
>                    ref="greetingService"/>
> ```
> 
> I found that there is no exposed service, but the container is started

config-1:
```xml
    <dubbo:provider export="false"/>
    <dubbo:service interface="com.code.dubbo.demo.DemoService" ref="demoServiceImpl"/>
```
result-1:
```text
[DUBBO] Export dubbo service com.code.dubbo.demo.DemoService to local registry url : injvm://127.0.0.1/com.code.dubbo.demo.DemoService?accesslog=true&....
```

config-2:
```xml
     <dubbo:provider export="false"/>
    <dubbo:service interface="com.code.dubbo.demo.DemoService" ref="demoServiceImpl" export="false"/>
```
result-2:
```text
not export
```

**Comment by gentryhuang** [comment]
> config:
> 
> ```
> <dubbo:protocol name="dubbo" port="-1"/>
> 
>     <dubbo:provider export="false" />
> 
>     <bean id="demoService" class="org.apache.dubbo.demo.provider.DemoServiceImpl"/>
>     <bean id="greetingService" class="org.apache.dubbo.demo.provider.GreetingServiceImpl"/>
> 
>     <dubbo:service interface="org.apache.dubbo.demo.DemoService" timeout="3000" ref="demoService" registry="registry1"/>
>     <dubbo:service version="1.0.0" group="greeting" timeout="5000" interface="org.apache.dubbo.demo.GreetingService"
>                    ref="greetingService"/>
> ```
> 
> I found that there is no exposed service, but the container is started

this issue has been resolved in 2.7.9
